### PR TITLE
Raise exception on permission-related rejection

### LINKED ISSUE #203: Publishes that fail due to permissions result in a TimeoutError
When using the twisted-backed publish API and the broker rejects a message because the user does not have permission to publish, the call times out instead of immediately raising a permission-related exception.

### PR DESCRIPTION
**Pull request by dvejmz**
Closes #203 

Raise permission exception if a publish via the
Twisted API was rejected due to insufficient
permissions

Signed-off-by: dvejmz <dvejmz@gmail.com>

### CODE REVIEW

**Review by jeremycline — DISMISSED** [review]
Generally this looks great, thanks for the pull request!

I'd really like to see an integration test to go along with it, there are a bunch of tests for the API in https://github.com/fedora-infra/fedora-messaging/blob/master/fedora_messaging/tests/integration/test_api.py.

**Comment by dvejmz** [comment]
@jeremycline sure, I'll write one, leave it with me!

**Comment by abompard** [comment]
Hey @dvejmz , are you still interested in writing this integration test?

**Comment by dvejmz** [comment]
Hi @abompard , I do apologise, this one definitely slipped through the cracks, I'd completely forgotten about this! I'll rebase this off master and write the integration test this week.

**Comment by abompard** [comment]
Hey! No problem. I recently fixed an issue that kinda looks similar but not entirely, so you can reuse the `PublishForbidden` exception I've added.

**Comment by dvejmz** [comment]
Hey guys @jeremycline @abompard , I've (finally!) now rebased this branch off the latest `stable` while making some changes to the original code I'd written:

* Changed the exception handler to raise the new `PublishForbidden` exception added by @abompard rather than the exception I had created
* Changed the `publish()` unit test I'd originally written to be more relevant
* Added the requested API integration test

Let me know if you have any other comments.

**Review by abompard — CHANGES_REQUESTED** [review]
Looks great! Could you please add a news fragment for the release notes? It's described [here](https://fedora-messaging.readthedocs.io/en/stable/contributing.html#release-notes). Thanks!

**Comment by dvejmz** [comment]
All done @abompard!

**Comment by abompard** [comment]
Thanks, but the `news/203.feature` file is empty. It should contain the snippet you want to appear in the release notes, for example "Raise PublishForbidden exception if publishing to channel is denied".

**Comment by dvejmz** [comment]
Ah, I see, I'm not familiar with the process, sorry about that. I'll get this fixed later today.

**Comment by dvejmz** [comment]
I've now updated the news fragment with the requested snippet. Thanks for your comments and patience! @abompard . I'll be sure to remember this next time.

File: news/203.feature [hunk 1]
```diff
@@ -0,0 +1 @@
+Raise `PublishForbidden` exception immediately if publishing to [virtual host](https://www.rabbitmq.com/vhosts.html) is denied rather than waiting until timeout.
```
Comment (abompard) [inline news/203.feature]: Thanks! Unfortunately this will be inserted in a RestructuredText document so this has to RST and not Markdown... sorry!

**Review by abompard — CHANGES_REQUESTED** [review]

File: news/203.feature (hunk 1 above)
Comment (dvejmz) [reply news/203.feature]: Fixed

**Review by abompard — APPROVED** [review]
Looks great, thanks a lot David!

**Pull request merged**